pkg/controller/keyvault/gateway_index.go

```diff
@@ -0,0 +1,70 @@
+package keyvault
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+func AddGatewayServiceAccountIndex(indexer client.FieldIndexer, indexName string) error {
+	if err := indexer.IndexField(context.Background(), &gatewayv1.Gateway{}, indexName, gatewayServiceAccountIndexFn); err != nil {
+		return fmt.Errorf("adding Gateway Service Account indexer: %w", err)
+	}
+
+	return nil
+}
+
+func gatewayServiceAccountIndexFn(object client.Object) []string {
+	gateway, ok := object.(*gatewayv1.Gateway)
+	if !ok {
+		return nil
+	}
+
+	saSet := map[string]struct{}{}
+
+	for _, listener := range gateway.Spec.Listeners {
```

Comment:
need to consider namespaces

Reply:
we consider namespaces when requeuing requests for Gateways - see line 60.

All we can do within this function is pull names from the Gateway resources - verification via the client naturally happens when we're grabbing the actual gateways tied to each serviceaccount